### DISCUSSION TITLE
Documentation error on how to use "--add-data" flag?

### DISCUSSION THREAD

**Discussion by MSLeiter** [pyinstaller/pyinstaller#6737] [Help] [answered]
The documentation for adding data files (See "Adding Files to Bundle" section here: https://github.com/pyinstaller/pyinstaller/blob/669313ba4c5c1403ebeb335c35cb68c8c6ba5dd4/doc/spec-files.rst) appears to have a syntax error, but just double checking here to make sure.

All the examples of how to use the "--add-data" (and further down, "--add-binary") flag at the command line use colons between the file-to-add and the location-to-add-it. For example:

```
pyinstaller --add-data 'src/README.txt:.' myscript.py
```

there is a colon between "src/README.txt" and ".". This is rejected by the pyi-makespec or pyinstaller commands at the command line. I found an example on StackOverflow that shows the syntax requires a semi-colon instead of a colon, and that appears to work fine. i.e. the above example becomes:

```
pyinstaller --add-data 'src/README.txt;.' myscript.py
```

Note that all the documentation on how to format the datas = [ ] tuples is fine, it's just the examples for the command-line version that appears to be wrong.

Does this mean the documentation is wrong?

**Comment by rokm** [comment, marked as answer]
The separator is OS-specific, same as `os.pathsep` - see https://pyinstaller.readthedocs.io/en/stable/usage.html#what-to-bundle-where-to-search that is also referenced at the section you linked.

The plan is to make it OS-independent, though (#6724).

**Reply by MSLeiter** [reply]
Okay, thanks - that makes sense. So it's treated like a PATH variable then.

Nice plan to update it. I commented there - using Python syntax itself (dict/list) would be nice, but with backwards compatibility and such might be better to just support both ; and :, etc.
